### ISSUE TITLE
9. Ensemble_CrossValidation

### ISSUE THREAD

**Issue by hih0124** [hih0124/lecture_python-II#45]
"""
교차검정(CrossValidation)
"""

from sklearn.datasets import load_digits # dataset
from sklearn.ensemble import RandomForestClassifier # model
from sklearn.model_selection import train_test_split, cross_validate # split, 교차검정
from sklearn.metrics import accuracy_score

### 1. dataset load
digit = load_digits()
X = digit.data
y = digit.target

X.shape  # (1797, 64)
y #  array([0, 1, 2, ..., 8, 9, 8])

### 2. model
rf = RandomForestClassifier()
model = rf.fit(X, y)

pred = model.predict(X) # class 예측치
pred # array([0, 1, 2, ..., 8, 9, 8])
pred2 = model.predict_proba(X)
pred2
'''
array([[1.  , 0.  , 0.  , ..., 0.  , 0.  , 0.  ],
       [0.  , 0.96, 0.01, ..., 0.  , 0.02, 0.  ],
       [0.  , 0.02, 0.84, ..., 0.  , 0.12, 0.  ],
       ...,
       [0.  , 0.07, 0.01, ..., 0.01, 0.85, 0.  ],
       [0.01, 0.01, 0.  , ..., 0.01, 0.04, 0.92],
       [0.  , 0.01, 0.06, ..., 0.  , 0.82, 0.01]])
'''

- 확률 -> index(10진수)
pred2_dit = pred2.argmax(axis = 1)
pred2_dit # array([0, 1, 2, ..., 8, 9, 8], dtype=int64) 최댓값의 인덱스를 반환해주는 것이것이 pred와 동일한 모양이 되는 것이다.

acc = accuracy_score(y,pred)
acc # 1.0

- CrossValidation
score = cross_validate(model, X, y, scoring='accuracy', cv = 5) # 5겹 검정
-이렇게 해도 되고, 스텝 1번처럼 모델 객체에 cv 넣어도 된다.
score
'''
{'fit_time': array([0.21316338, 0.20461392, 0.21456742, 0.21210814, 0.21396995]),
 'score_time': array([0.01020479, 0.01025009, 0.01033378, 0.01010728, 0.01012278]),
 'test_score': array([0.93888889, 0.91111111, 0.9637883 , 0.9637883 , 0.91643454])}
test_score ( 1겹 교차검정에서의 정확도),(2겹교차검정에서의 정확도)
'''
score
test_score = score['test_score']
test_score.mean() # 0.9388022284122564 평균 정확도
